src/app/views/realms/view/view.store.ts

```diff
@@ -0,0 +1,42 @@
+import { inject, InjectionKey, ref, Ref } from 'vue';
+import { Fireside } from '../../../../_common/fireside/fireside.model';
+import { ModelClassType } from '../../../../_common/model/model.service';
+import { Realm } from '../../../../_common/realm/realm-model';
+import { User } from '../../../../_common/user/user.model';
+
+export const RealmRouteStoreKey: InjectionKey<RealmRouteStore> = Symbol('realm-store');
+
+export type RealmRouteStore = ReturnType<typeof createRealmRouteStore>;
+
+export type RealmRoutePayload = {
+	realm: ModelClassType<Realm>;
+	knownFollowers: ModelClassType<User>[];
+	knownFollowerCount: number;
+	activeFiresides: ModelClassType<Fireside>[];
+};
+
+export function createRealmRouteStore() {
+	const realm = ref() as Ref<Realm>;
```

Comment:
Is there a reason we're not defining this as `ref<Realm>()` so it's nullable? It looks like a lot of places that access this are checking for nullability anyway.

Reply:
i think i did this when my volar was broken hah. ill check.